lib/DSA/DSGraph.cpp

```diff
@@ -1091,8 +1091,11 @@ void DSGraph::removeDeadNodes(unsigned Flags) {
 
   // The return values are alive as well.
   for (ReturnNodesTy::iterator I = ReturnNodes.begin(), E = ReturnNodes.end();
-       I != E; ++I)
-    I->second.getNode()->markReachableNodes(Alive);
+       I != E; ++I) {
+    auto node = I->second.getNode();
+    if (node)
```

Comment:
This link appears to reiterate the fact that undefined behaviors are triggered. So invoking this class method seems to be undefined and should be avoided.

Reply:
Yes, that is undefined behavior. But program sometimes rely on compilers doing the usual thing, like that webpage describes. DSA is one such program - I remember that it invokes a function through a null pointer and it expects for that function to get invoked (which is what happens in most modern compilers).
Like I said, I did the same fixes as you several years ago (to fix these warnings), only to learn that I broke DSA since it requires that a function gets invoked on null pointer. Does this make sense?
I am ok that we try to do these fixes again, but but them in a separate pull request since I want that we thoroughly test them on SVCOMP benchmarks.

Reply:
Sounds good.

Reply:
Ok. I reverted the commit.